clients/fides-js/src/components/Overlay.tsx

```diff
@@ -34,6 +38,11 @@ interface Props {
   renderModalContent: (props: RenderModalContent) => VNode;
 }
 
+const getLatestCookie = (cookieFromParam: FidesCookie) => {
```

Comment:
oh right I don't think I ever explained why this was here... I can add a comment, but the idea was that the `cookie` passed into `Overlay` doesn't update when the cookie updates. We need the updated cookie to pass along in the various `FidesUIShown` and `FidesModalClosed` events, so this attempts to get the latest cookie. I think the only reason why we wouldn't prefer the cookie on the DOM is if there is no cookie on the DOM yet, but this logic could use a second pair of eyes!